unica: patches: vndk: support for devices without vndk

Fede2782 opened the pull request.

Fede2782 committed:
unica: patches: vndk: support for devices without vndk

Co-authored-by: ExtremeXT <75576145+ExtremeXT@users.noreply.github.com>

nhglong commented on unica/patches/vndk/customize.sh:
```diff
@@ -1,23 +1,39 @@
-                ;;
-        esac
+        ADD_TARGET_VNDK_APEX
         sed -i "s/version>$SOURCE_VNDK_VERSION/version>$TARGET_VNDK_VERSION/g" "$SYS_EXT_DIR/etc/vintf/manifest.xml"
```
is this line needed even

fivemeyestoreserver commented:
Interesting solution to this problem! I've been working on similar challenges recently.

Fede2782 committed:
unica: patches: vndk: rename from VNDK_VERSION to VENDOR_API_LEVEL

Android 15 deprecates VNDK. This also makes more clear and allows the usage
of one variable for multiple future usages.
Vendor API >= 35 means not VNDK.
Vendor API <= 34 means VNDK.

Change-Id: Ie8d9bd4e07927738a21a78cb732f011d67707ac7

salvogiangri committed:
unica: patches: vndk: require VENDOR_API_LEVEL to be set

Change-Id: I76292096480d595e40287f7a060586b94381151b

salvogiangri committed:
unica: patches: vndk: minor adjustments

Change-Id: I8aab8c8ce5109867c9593cddf74630cc19bf152c
